File: CODIGOS_PYTHON/ACT5.py
class Objeto:  #Define la clase para los objetos que se agregaran a la coleccion
    def __init__(self, identificador, atributo):
        self.identificador = identificador
        self.atributo = atributo

    def get_identificador(self):
        return self.identificador

    def get_atributo(self):
        return self.atributo

    def set_atributo(self, atributo):
        self.atributo = atributo


class EntidadPrincipal:  #Clase base: Entidad principal en la relacion de agregacion
    def __init__(self):
        self.coleccion = []  #Lista que representa la coleccion en la que se agregaron los objetos

    def agregar_objeto(self, objeto):
        self.coleccion.append(objeto)

    def eliminar_objeto(self, objeto):
        for obj in self.coleccion:
            if obj.get_identificador() == objeto.get_identificador():
                self.coleccion.remove(obj)
                return
        print("El objeto no está en la colección.")

    def obtener_objeto(self, identificador):
        for obj in self.coleccion:
            if obj.get_identificador() == identificador:
                return obj

    def get_coleccion(self):
        return self.coleccion


if __name__ == "__main__":
    entidad_principal = EntidadPrincipal()

    #Crear varios objetos de la clase que se agregaron a la coleccion
    objeto1 = Objeto(1, "Atributo 1")
    objeto2 = Objeto(2, "Atributo 2")
    objeto3 = Objeto(3, "Atributo 3")

    #Agregar los objetos a la coleccion
    entidad_principal.agregar_objeto(objeto1)
    entidad_principal.agregar_objeto(objeto2)
    entidad_principal.agregar_objeto(objeto3)

    #Obtener un objeto especifico de la coleccion mediante su identificador
    objeto = entidad_principal.obtener_objeto(2)
    if objeto:
        print("Identificador:", objeto.get_identificador(), ", Atributo:", objeto.get_atributo())

    #Eliminar un objeto de la coleccion
    entidad_principal.eliminar_objeto(objeto3)
    print("Objeto eliminado.")

    #Verificar los objetos en la coleccion despues de eliminar uno
    print("Objetos en la colección después de eliminar:")
    for obj in entidad_principal.get_coleccion():
        print("Identificador:", obj.get_identificador(), ", Atributo:", obj.get_atributo())

File: CODIGOS_JAVA/ACT5.java
import java.util.ArrayList; // Importa la clase ArrayList para trabajar con colecciones dinámicas
import java.util.List;      // Importa la clase List para trabajar con interfaces de colecciones

class Objeto {              // Define la clase para los objetos que se agregarán a la colección
    private int identificador;
    private String atributo;

    public Objeto(int id, String attr) {  // Constructor con dos parámetros
        identificador = id;
        atributo = attr;
    }
    public int getIdentificador() {       // Método que devuelve el identificador de un objeto
        return identificador;
    }
    public String getAtributo() {         // Método que devuelve el atributo de un objeto
        return atributo;
    }
    public void setAtributo(String attr) {// Método para establecer el atributo del objeto en una variable
        atributo = attr;
    }
}
class EntidadPrincipal {                // Clase base: Entidad principal en la relación de agregación
    private List<Objeto> coleccion;     // Lista que representa la colección en la que se agregarán los objetos

    public EntidadPrincipal() {         // Constructor que inicializa la colección
        coleccion = new ArrayList<>();
    }
    public void agregarObjeto(Objeto objeto) {  // Método para agregar un objeto a la colección de objetos
        coleccion.add(objeto);
    }
    public void eliminarObjeto(Objeto objeto) { // Método que recibe un objeto y lo elimina de la colección
        coleccion.remove(objeto);
    }
    public Objeto obtenerObjeto(int identificador) {  // Método para obtener un objeto por su identificador
        for (Objeto obj : coleccion) {
            if (obj.getIdentificador() == identificador) {
                return obj;
            }
        }
        return null;
    }
    public List<Objeto> getColeccion() {  // Método que devuelve una referencia a la colección de objetos
        return coleccion;
    }
}

public class Act5 {
    public static void main(String[] args) {
        EntidadPrincipal entidadPrincipal = new EntidadPrincipal();

        // Crear varios objetos de la clase que se agregarán a la colección
        Objeto objeto1 = new Objeto(1, "Atributo 1");
        Objeto objeto2 = new Objeto(2, "Atributo 2");
        Objeto objeto3 = new Objeto(3, "Atributo 3");
        // Agregar los objetos a la colección
        entidadPrincipal.agregarObjeto(objeto1);
        entidadPrincipal.agregarObjeto(objeto2);
        entidadPrincipal.agregarObjeto(objeto3);
        // Obtener un objeto específico de la colección mediante su identificador
        Objeto objeto = entidadPrincipal.obtenerObjeto(2);
        if (objeto != null) {
            System.out.println("Identificador: " + objeto.getIdentificador() + ", Atributo: " + objeto.getAtributo());
        }
        // Eliminar un objeto de la colección
        entidadPrincipal.eliminarObjeto(objeto3);
        System.out.println("Objeto eliminado.");
        // Verificar los objetos en la colección después de eliminar uno
        System.out.println("Objetos en la colección después de eliminar:");
        for (Objeto obj : entidadPrincipal.getColeccion()) {
            System.out.println("Identificador: " + obj.getIdentificador() + ", Atributo: " + obj.getAtributo());
        }
    }
}
